### DISCUSSION TITLE
报错了这个怎么办。

### DISCUSSION THREAD

**Discussion by shmilyabyy** [nxtrace/NTrace-core#67] [Q&A] [answered]
fatal error: concurrent map read and map write。

**Comment by tsosunchia** [comment, marked as answer]
请提供更详细的命令

**Reply by shmilyabyy** [reply]
![image](https://user-images.githubusercontent.com/77541484/215330069-3551a4c2-36e0-4ed8-a19e-6a19d74796b1.png)
![image](https://user-images.githubusercontent.com/77541484/215330082-3bc062b6-a0ab-4414-93fe-adddec2e2f02.png)

**Reply by shmilyabyy** [reply]
但是我在其他地区的机器上却没事，我测试了三台美国机器都是这样。

**Reply by tsosunchia** [reply]
`101.226.28.198` 您可以ping通吗？

**Reply by shmilyabyy** [reply]
不可以

**Reply by tsosunchia** [reply]
这是您网络的问题，不是nexttrace软件的问题。
请联系您的网络提供商，以尝试解决。
（顺便说一下，大部分国外VPS是不保证到大陆路由的）

**Reply by shmilyabyy** [reply]
我是用机器去ping他是吗，我一台悉尼的机器也ping不通，但是可以显示回程路由

**Reply by sjlleo** [reply]
> 我是用机器去ping他是吗，我一台悉尼的机器也ping不通，但是可以显示回程路由

Hi shmilyabyy,

您有安装 mtr 吗，为了更好的排查故障所在，可以麻烦您可以使用 mtr 运行以下命令吗？

```
mtr -r 106.37.67.1
```

谢谢！

**Reply by shmilyabyy** [reply]
![image](https://user-images.githubusercontent.com/77541484/215331116-f2ac6fd0-3f69-44b4-a299-107ac90ddcb6.png)
![image](https://user-images.githubusercontent.com/77541484/215331147-558ec880-fe4a-483d-9463-160bda2f9c09.png)
这是澳洲的机器同样是ping不通这个ip但是没有报错

**Reply by shmilyabyy** [reply]
> > 我是用机器去ping他是吗，我一台悉尼的机器也ping不通，但是可以显示回程路由
> 
> Hi shmilyabyy,
> 
> 您有安装 mtr 吗，为了更好的排查故障所在，可以麻烦您可以使用 mtr 运行以下命令吗？
> 
> ```
> mtr -r 106.37.67.1
> ```
> 
> 谢谢！

![image](https://user-images.githubusercontent.com/77541484/215331354-d626768a-900b-4fea-8725-24d0cf52d748.png)
![image](https://user-images.githubusercontent.com/77541484/215331363-fec133bc-78e7-489c-89bb-acac71da072d.png)
这是两台美国的机器的mtr

**Reply by tsosunchia** [reply]
我拿我的香港xtom(V.PS)机器也试了一下，是正常的

**Reply by tsosunchia** [reply]
<img width="690" alt="image" src="https://user-images.githubusercontent.com/59512455/215332170-24222a2e-949a-49ff-a9dd-7c79ab8d18bc.png">

**Reply by sjlleo** [reply]
> > > 我是用机器去ping他是吗，我一台悉尼的机器也ping不通，但是可以显示回程路由
> > 
> > 
> > Hi shmilyabyy,
> > 您有安装 mtr 吗，为了更好的排查故障所在，可以麻烦您可以使用 mtr 运行以下命令吗？
> > ```
> > mtr -r 106.37.67.1
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 谢谢！
> 
> ![image](https://user-images.githubusercontent.com/77541484/215331354-d626768a-900b-4fea-8725-24d0cf52d748.png) ![image](https://user-images.githubusercontent.com/77541484/215331363-fec133bc-78e7-489c-89bb-acac71da072d.png) 这是两台美国的机器的mtr

这看起来是在圣何塞的 xTom 网络中，不知何缘故，ICMP 包的 seq 丢失了导致的，一般来说这种情况很罕见，通常路由不会丢弃这些信息，因为这些信息的丢失或多或少会导致一些程序异常。
seq 信息对 NT 很重要，由于NT 没有使用原生系统 Library，所以 ICMP 应答中的 seq 的缺失可能会导致包无法正常被解析
这种情况，我一般推荐改用 TCP 进行路由跟踪，应该可以彻底解决这个问题。

谢谢

**Reply by shmilyabyy** [reply]
好的

**Reply by shmilyabyy** [reply]
> 我拿我的香港xtom(V.PS)机器也试了一下，是正常的

我的香港也可以显示回程，但是ping不通

**Reply by sjlleo** [reply]
> 好的

如果这个问题已经解决，我将关闭这个讨论，如果您有其他任何问题，欢迎再来提问。

**Reply by sjlleo** [reply]
> 好的

我昨天晚上研究了一下，现在的 1.0.6 版本应该可以解决这个问题，试试哦

**Comment by haima3** [comment]
> fatal error: concurrent map read and map write。

请提供可复现的方式，您命令是怎样写的呢？有截图吗？

**Reply by shmilyabyy** [reply]
![image](https://user-images.githubusercontent.com/77541484/215330230-4f1b5fc2-55d6-486b-983b-595bc2540309.png)
![image](https://user-images.githubusercontent.com/77541484/215330239-a40eda77-0103-40d8-b493-a6577a6b6cdd.png)
